examples/sync/README.md

```diff
@@ -0,0 +1,110 @@
+# ADB Sync Example
+
+In this example, we synchronize a new [adb::any::Any](../../storage/src/adb/any/mod.rs) database to the server's database state.
+
+## Components
+
+- [Server](src/bin/server.rs): Serves historical operations and proofs to clients.
+- [Client](src/bin/client.rs): Starting from an empty database, syncs to the server's database state.
+- [NetworkResolver](src/resolver.rs): Used by client to communicate with server.
+- [Protocol](src/protocol.rs): Defines network messages.
+
+## Build
+
+```bash
+cargo build
+```
+
+## Run Tests
+
+```bash
+cargo test
+```
+
+## Usage
+
+### Running the Server
+
+```bash
+# Start server with default settings (port 8080, 100 initial operations)
+cargo run --bin server
+
+# Start server with custom settings
+cargo run --bin server -- --port 8080 --initial-ops 50 --storage-dir /tmp/my_server --seed 1337 --metrics-port 9091
+```
+
+Server options:
+- `-p, --port <PORT>`: Port to listen on (default: 8080)
+- `-i, --initial-ops <COUNT>`: Number of initial operations to create (default: 100)
+- `-d, --storage-dir <PATH>`: Storage directory (default: /tmp/adb_sync_server)
+- `-s, --seed <SEED>`: Seed for generating test operations (default: 1337)
+- `-m, --metrics-port <PORT>`: Port on which metrics are exposed (default: 9091)
+
+### Running the Client
+
+```bash
+# Connect to server with default settings
+cargo run --bin client
+
+# Connect with custom settings
+cargo run --bin client -- --server 127.0.0.1:8080 --batch-size 25 --storage-dir /tmp/my_client --metrics-port 9090
+```
+
+Client options:
```

Comment:
To better approximate reality, I'm wondering if we should require the client to provide the "sync target" (so its clear you aren't supposed to "trust" the server)?